packages/svelte/src/compiler/compile/render_dom/invalidate.js

```diff
@@ -61,8 +62,9 @@ export function invalidate(renderer, scope, node, names, main_execution_context
 		return x`@set_store_value(${head.name.slice(1)}, ${node}, ${head.name}, ${extra_args})`;
 	}
 
+	const is_props = !!head.export_name;
 	let invalidate;
-	if (!main_execution_context) {
+	if (!main_execution_context || is_props) {
```

Comment:
Compiler also inserts `$$invalidate` around exported variables at instance `main_execution_context`.

Reply:
@Conduitry / @tanhauhau is this ok to add invalidation in this case, too? Why's invalidation generally forbidden in the main execution context?